src/commands/createFunctionApp/FunctionAppHostingPlanStep.ts

```diff
@@ -3,26 +3,37 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AppServicePlanListStep, setLocationsTask } from '@microsoft/vscode-azext-azureappservice';
+import { AppServicePlanListStep, WebsiteOS, setLocationsTask, type IAppServiceWizardContext } from '@microsoft/vscode-azext-azureappservice';
+import { LocationListStep } from '@microsoft/vscode-azext-azureutils';
 import { AzureWizardPromptStep, type IAzureQuickPickItem, type IWizardOptions } from '@microsoft/vscode-azext-utils';
 import { localize } from '../../localize';
 import { getRandomHexString } from '../../utils/fs';
 import { nonNullProp } from '../../utils/nonNull';
+import { getWorkspaceSetting } from '../../vsCodeConfig/settings';
 import { type IFunctionAppWizardContext } from './IFunctionAppWizardContext';
 
 export class FunctionAppHostingPlanStep extends AzureWizardPromptStep<IFunctionAppWizardContext> {
     public async prompt(context: IFunctionAppWizardContext): Promise<void> {
         const placeHolder: string = localize('selectHostingPlan', 'Select a hosting plan.');
+        const enableFlexSetting = await getWorkspaceSetting('enableFlexConsumption');
         const picks: IAzureQuickPickItem<[boolean, RegExp | undefined]>[] = [
             { label: localize('consumption', 'Consumption'), data: [true, undefined] },
             { label: localize('premium', 'Premium'), data: [false, /^EP$/i] },
             { label: localize('dedicated', 'App Service Plan'), data: [false, /^((?!EP|Y).)*$/i] }
         ];
 
+        if (enableFlexSetting) {
+            picks.splice(1, 0, { label: localize('flexConsumption', 'Flex Consumption (Preview)'), data: [false, undefined] });
+        }
+
         [context.useConsumptionPlan, context.planSkuFamilyFilter] = (await context.ui.showQuickPick(picks, { placeHolder })).data;
         await setLocationsTask(context);
         if (context.useConsumptionPlan) {
             setConsumptionPlanProperties(context);
+        } else if (!context.useConsumptionPlan && !context.planSkuFamilyFilter) {
+            // if it's not consumption and has no filter, then it's flex consumption
+            setFlexConsumptionPlanProperties(context);
+            await LocationListStep.setLocation(context, 'North Central US (Stage)');
```

Comment:
Are you hardcoding this value for test purposes / Does this need to be updated before you merge this PR or is it a future TODO?

Reply:
Yes, there is a list of locations in the `constants.ts` that I'll have to use later. But as of now, there's no other location that works for creating this flex app service plan.